src/App.tsx

```diff
@@ -1,26 +1,222 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { createTodos, deleteTodo, getTodos, USER_ID } from './api/todos';
+
+import { ErrorMessage } from './types/ErrorMessage';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { TodoFooter } from './components/TodoFooter';
+import { TodoItem } from './components/TodoItem';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoHeader } from './components/TodoHeader';
+import { Status } from './types/Status';
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
 export const App: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>(
+    ErrorMessage.Default,
+  );
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeStatus, setActiveStatus] = useState(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoadingTodos, setIsLoadingTodos] = useState<number[]>([]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const filteredTodos = getFilteredTodos(todos, activeStatus);
+  const notCompletedTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const onAddTodo = async (title: string) => {
+    setTempTodo({
+      id: 0,
+      title,
+      userId: USER_ID,
+      completed: false,
+    });
+
+    const newTodo: Omit<Todo, 'id'> = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    try {
+      const todo = await createTodos(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, todo]);
+    } catch (error) {
+      setErrorMessage(ErrorMessage.UnableToAdd);
+      throw error;
```

Comment:
You don't need to throw an error if you don't plane to handle it in other block

Reply:
in this case I catch error in handleSubmit in TodoHeader component